pkg/factsengine/entities/fact_value.go

```diff
@@ -44,20 +57,34 @@ func NewFactValue(factInterface interface{}) (FactValue, error) {
 	case map[string]interface{}:
 		newMap := make(map[string]FactValue)
 		for key, mapValue := range value {
-			newValue, err := NewFactValue(mapValue)
+			newValue, err := NewFactValueWithConf(mapValue, conf)
 			if err != nil {
 				return nil, err
 			}
+			if conf.SnakeCaseKeys {
+				key = strcase.ToSnake(key)
```

Comment:
Can we use a new scoped variable, not assigning to the key moving pointer in the for loop?